src/traces/heatmap/hover.js

```diff
@@ -96,8 +96,8 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode, contour)
     if(zmask && !zmask[ny][nx]) zVal = undefined;
 
     var text;
-    if(Array.isArray(trace.text) && Array.isArray(trace.text[ny])) {
-        text = trace.text[ny][nx];
+    if(Array.isArray(cd0.text) && Array.isArray(cd0.text[ny])) {
+        text = cd0.text[ny][nx];
```

Comment:
as for xyz heatmaps, `fullData[i].trace.text` is 1D, whereas `gd.calcdata[i][0].text` is 2D.

Previously, `fullData[i].trace.text` was converted to be 2D during the calc step, but since the calc step isn't called on `relayout(gd, 'xaxis.range', [/* */])`, `text` on this line here was `undefined` after zoom interactions as discovered in https://github.com/plotly/plotly.js/issues/1391.